shared/fs/footer/container.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import * as I from 'immutable'
+import * as Types from '../../constants/types/fs'
+import {compose, connect, setDisplayName, type Dispatch, type TypedState} from '../../util/container'
+import Footer, {type FooterProps} from './footer'
+import * as FsGen from '../../actions/fs-gen'
+
+type StateProps = {
+  transfers: I.Map<string, Types.TransferState>,
+}
+
+type OwnProps = {}
+
+type actionFactory = (p: Types.LocalPath) => () => void
+type DispatchProps = {
+  opener: actionFactory,
+  dismisser: actionFactory,
+}
+
+const mapStateToProps = (state: TypedState, ownProps: OwnProps) => ({
+  transfers: state.fs.transfers,
+})
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  opener: (p: Types.LocalPath) => () =>
+    dispatch(
+      FsGen.createOpenLocalPathInFolder({
+        localPath: p,
+      })
+    ),
+  dismisser: (key: string) => () => dispatch(FsGen.createDismissTransfer({key})),
+})
+
+const mergeProps = (stateProps: StateProps, {opener, dismisser}: DispatchProps, ownProps) =>
+  ({
+    downloads: Array.from(stateProps.transfers.filter(transferState => transferState.type === 'download'))
```

Comment:
do we need to do all this work here? should the representation in the store be simpler so the containers don't have to do this kind of mapping

Reply:
You mean this particular `filter` call or this entire thing including mapping? If `filter`, I guess we could have separate `downloads`/`uploads` in store.